src/bin/main.rs

```diff
@@ -70,19 +95,68 @@ async fn main() {
     tracing::subscriber::set_global_default(subscriber)
         .expect("Failed on subscribe tracing");
 
-    // Startup call sending the possible args passed
-    match startup(
-        matches.value_of("host").unwrap_or_else(|| HOST_ADDRESS),
-        matches.value_of("port").unwrap_or_else(|| PORT),
-    )
-    .await
-    {
-        Ok(_) => (),
+    // Match the desired IPC method. Or set the default one depending on the OS used.
+    // Then startup rusk with the final values.
+    let res = match matches.value_of("ipc_method") {
+        Some(method) => match (cfg!(windows), method) {
+            (_, "tcp_ip") => {
+                startup_with_tcp_ip(
+                    matches.value_of("host").unwrap_or_else(|| HOST_ADDRESS),
+                    matches.value_of("port").unwrap_or_else(|| PORT),
+                )
+                .await
+            }
+            (true, "uds") => {
+                panic!("Windows does not support Unix Domain Sockets");
```

Comment:
Should be `unreachable` instead of `panic` in case the previous comment is fixed

Reply:
Probably not going to modify the previous unless it's something you consider critical.